src/SFA.DAS.Employer.PR.Web/Controllers/SelectTrainingProviderController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using FluentValidation;
+using FluentValidation.AspNetCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Employer.PR.Web.Authentication;
+using SFA.DAS.Employer.PR.Web.Infrastructure;
+using SFA.DAS.Employer.PR.Web.Infrastructure.Services;
+using SFA.DAS.Employer.PR.Web.Models;
+using SFA.DAS.Employer.PR.Web.Models.Session;
+
+namespace SFA.DAS.Employer.PR.Web.Controllers;
+
+[Authorize(Policy = nameof(PolicyNames.HasEmployerOwnerAccount))]
+[Route("accounts/{employerAccountId}/providers/new/selectProvider", Name = RouteNames.SelectTrainingProvider)]
+public class SelectTrainingProviderController(ISessionService _sessionService, IValidator<SelectTrainingProviderSubmitViewModel> _validator) : Controller
+{
+    [HttpGet]
+    public IActionResult Index([FromRoute] string employerAccountId)
+    {
+        var sessionModel = _sessionService.Get<AddTrainingProvidersSessionModel>();
+
+        if (sessionModel == null)
+        {
+            return RedirectToAction("Index", "YourTrainingProviders", new { employerAccountId });
+        }
+
+        var backLink = SetBackLink(employerAccountId, sessionModel.AccountLegalEntities.Count);
+
+        SelectTrainingProviderViewModel model = new SelectTrainingProviderViewModel(backLink!, sessionModel!.ProviderName, sessionModel!.Ukprn.ToString());
+        return View(model);
+    }
+
+    [HttpPost]
+    public IActionResult Index([FromRoute] string employerAccountId, SelectTrainingProviderSubmitViewModel submitModel)
+    {
+        var sessionModel = _sessionService.Get<AddTrainingProvidersSessionModel>();
+        submitModel.SearchTerm = submitModel.Name;
```

Comment:
What is this for?

Reply:
its because the text box needs to have a distinct name from the actual value to allow the validation flow to work.  'Name' is the actual stored value (along with ukprn), but the validator works against the name of the text control, which is 'SearchTerm'.  We've used this pattern elsewhere.